uamqp/async_ops/connection_async.py

```diff
@@ -98,6 +99,15 @@ def __init__(self, hostname, sasl,
         await self.destroy_async()
         _logger.debug("Finished exiting connection %r context.", self.container_id)
 
+    async def _open_async(self):
```

Comment:
Should we allow for a timeout to be passed? (re: potentially looping forever)

Reply:
you're right, I need to check whether the underlying c/cython code would take care of the open timeout. If not then I need to add check here.

Reply:
I checked the underlying socketio_dowork but find nothing related to timeout control.
added timeout control here.